src/main/java/wis/util/InputParser.java

```diff
@@ -0,0 +1,136 @@
+    }
+
+    /**
+     * Parses user input for description of todo task.
+     *
+     * @param input Line of user input.
+     * @return Description of todo task.
+     * @throws InputMismatchException  If parsing fails as the line has
+     * unknown format.
+     */
+    public static String parseTodo(String input) throws InputMismatchException {
+        Pattern pattern = Pattern.compile("^todo ([^ ].*)$");
+        Matcher matcher = pattern.matcher(input);
+        if (!matcher.find()) {
+            throw new InputMismatchException();
+        }
+        return matcher.group(1);
+    }
+
+    private static boolean isSingleOccurrence(String pattern, String input) {
+        String[] parts = input.split(pattern);
+        return parts.length == 2;
+    }
+
+    /**
+     * Parses user input for description and deadline of deadline task.
+     *
+     * @param input Line of user input.
+     * @return String Array containing description and deadline of deadline task.
+     * @throws InputMismatchException  If parsing fails as the line has
+     * unknown format.
+     */
+    public static String[] parseDeadline(String input) {
+        String[] ret = new String[3];
+        Pattern pattern = Pattern.compile("^deadline ([^ ].*) /by ([^ ].*)$");
+        Matcher matcher = pattern.matcher(input);
+        if (matcher.find() && isSingleOccurrence("/by", input)) {
+            ret[0] = matcher.group(1);
+            ret[1] = matcher.group(2);
+        } else {
+            throw new InputMismatchException("Unknown input format");
+        }
+        return ret;
+    }
+
+    /**
+     * Parses user input for description of event task.
+     *
+     * @param input Line of user input.
+     * @return String Array containing description, start time and end time of
+     * deadline task.
+     * @throws InputMismatchException  If parsing fails as the line has
+     * unknown format.
+     */
+    public static String[] parseEvent(String input) {
+        String[] ret = new String[3];
+        Pattern pattern = Pattern.compile("^event ([^ ].*) /from ([^ ].*) /to ([^ ].*)$");
+        Matcher matcher = pattern.matcher(input);
+        if (matcher.find() &&
+                isSingleOccurrence("/from", input) &&
+                isSingleOccurrence("/to", input)) {
+            ret[0] = matcher.group(1);
+            ret[1] = matcher.group(2);
+            ret[2] = matcher.group(3);
+        } else {
+            throw new InputMismatchException("Unknown input format");
+        }
+        return ret;
+    }
+
+    /**
+     * Parses date and time input strings to return a Java LocalDateTime object.
+     *
+     * @param string Strings of the form yyyy-mm-dd hh:mm.
+     * @return A Java LocalDateTime object.
+     */
+    public static LocalDateTime parseDateTime(String string)
+            throws DateTimeParseException, ArrayIndexOutOfBoundsException {
+        // Input format: 2023-12-03 10:15
+        String[] strings = string.split(" ");
```

Comment:
a bit of nitpick
perhaps
```suggestion
        String[] wordsInString = string.split(" ");
```